lib/bundler/lockfile_parser.rb

```diff
@@ -171,43 +171,45 @@ def parse_source(line)
       end
     end
 
-    NAME_VERSION = '(?! )(.*?)(?: \(([^-]*)(?:-(.*))?\))?'.freeze
-    NAME_VERSION_2 = /^ {2}#{NAME_VERSION}(!)?$/
-    NAME_VERSION_4 = /^ {4}#{NAME_VERSION}$/
-    NAME_VERSION_6 = /^ {6}#{NAME_VERSION}$/
+    NAME_VERSION = /^( {2}| {4}| {6})(?! )(.*?)(?: \(([^-]*)(?:-(.*))?\))?(!)?$/
```

Comment:
Also, for below, would it make sense to use named capture groups? That way we don't have to just match at `$1`, it could be `match['name']`, `match['pinned']`